### PR TITLE
feat(string): nanoid

### LINKED ISSUE #1426: Add nanoid for uuid alternatives
### Clear and concise description of the problem

nanoid is a great alternative for uuid, it is had less complexity and size.

it is commonly used for blog post unique urls.

### Suggested solution

Add nanoid, the logic should based on https://github.com/ai/nanoid

### Alternative

No alternative for https://github.com/ai/nanoid

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by xDivisionByZerox**
Resolves #1426.

- #1426

### CODE REVIEW

File: test.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { faker } from './src';
+
+console.log(faker.string.nanoid());
+console.log(faker.string.nanoid(10));
+console.log(faker.string.nanoid({ min: 13, max: 37 }));
```
Comment (ST-DDT) [inline test.ts]: Please remove this debug code.

File: src/modules/string/index.ts [hunk 2]
```diff
@@ -532,6 +532,50 @@ export class StringModule {
...
+   *
+   * @since 8.0.0
+   */
+  nanoid(length: number | { min: number; max: number } = 21): string {
+    length = this.faker.helpers.rangeToNumber(length);
+    if (length <= 0) {
+      return '';
+    } else if (length > 36) {
+      length = 36;
+    }
```
Comment (ST-DDT) [inline src/modules/string/index.ts]: The original implementation doesn't seem to have this limit:

https://github.com/ai/nanoid/blob/6cd578dca8c5c45946eb15796de3e3c91fa187c4/non-secure/index.js#L23-L32

**Review by ST-DDT — CHANGES_REQUESTED** [review]

File: test.ts (hunk 1 above)
Comment (xDivisionByZerox) [reply test.ts]: ofc, my bad.

File: src/modules/string/index.ts (hunk 2 above)
Comment (xDivisionByZerox) [reply src/modules/string/index.ts]: I got that from their [official website](https://zelark.github.io/nano-id-cc/), where `36` is the maximum size of an id.

Comment (ST-DDT) [reply src/modules/string/index.ts]: > More than 1 quadrillion years needed, in order to have a 1% probability of at least one collision.

I assume that after that the math no longer works out regarding their collision detection.
(that page is specifically for collision probability calculation)

Their source code doesn't have a limit so we shouldn't have one either IMO.

Comment (xDivisionByZerox) [reply src/modules/string/index.ts]: No problem. I removed the maximum restriction.

**Review by ST-DDT — APPROVED** [review]

**Review by ejcheng — APPROVED** [review]

File: src/modules/string/index.ts [hunk 3]
```diff
@@ -532,6 +532,48 @@ export class StringModule {
...
+    const generators = [
+      {
+        value: () => this.alphanumeric(1),
+        // a-z is 26 characters
+        // this times 2 for upper & lower case is 52
+        // add all numbers 0-9 (10 in total) you get 62
+        weight: 62,
+      },
+      {
+        value: () => this.faker.helpers.arrayElement(['_', '-']),
```
Comment (Shinigami92) [inline src/modules/string/index.ts:564]: I searched if it's allowed to start with `_|-` but I found nothing :shrug: 
So seems it's okay

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**